client/src/components/Table/renderers.tsx

```diff
@@ -154,3 +156,46 @@ export const scoreRenderer = (item: CourseScheduleItemDto) => {
     </Text>
   );
 };
+
+export const renderTask = (name: string, item: CourseScheduleItemDto) => {
+  if (!item.descriptionUrl) return name;
+
+  return (
+    <Link target="_blank" href={item.descriptionUrl}>
+      {name}
+    </Link>
+  );
+};
+
+export const coloredDateRenderer = (timeZone: string, format: string, date: 'start' | 'end', infoText: string) => {
+  const now = moment();
+  return (value: string, { startDate, endDate, score, tag }: CourseScheduleItemDto) => {
+    let color: BaseType | undefined = undefined;
+    const start = moment(startDate);
+    const end = moment(endDate);
+
+    const isDeadlineSoon = now <= end && end.diff(now, 'hours') < 48 && !score;
+    const isCurrent = now >= start && now < end && !score;
+    const isDeadlineMissed = now >= end && end.diff(now, 'hours') >= -24 && !score;
+    const isPast = now > end || score;
+
+    if (isDeadlineSoon && date === 'end') color = 'warning';
+    else if (isCurrent && date === 'start') color = 'success';
+    else if (isDeadlineMissed && date === 'end') color = 'danger';
+    else if (isPast) color = 'secondary';
```

Comment:
we do prefer brackets even for single line statements.
consider moving  the whole block to function with return statements

Reply:
It was moved as-is from TableView renderers for reuse on Student Dashboard. I haven't done any change